tests/models/UserProfilingModel.spec.ts

```diff
@@ -131,6 +131,25 @@ describe('UserProfilingModel', () => {
     });
 
     describe('getActions', () => {
+        let logSearchEventStub: SinonStub;
+        let sendSearchEventsStub: SinonStub;
+
+        beforeEach(() => {
+            sandbox.stub(Component, 'get').callsFake(() => {
+                logSearchEventStub = sandbox.stub();
+                sendSearchEventsStub = sandbox.stub();
+                return {
+                    usageAnalytics: {
+                        logSearchEvent: logSearchEventStub,
+                        getPendingSearchEvent: sandbox.stub().returns({}),
+                        endpoint: {
+                            sendSearchEvents: sendSearchEventsStub,
+                        },
+                    },
+                } as any;
```

Comment:
Ain't that at least a Partial something?

Reply:
That's a test mock polyfill